### PR TITLE
Hw4 branch

### PR DESCRIPTION
**Pull request by juliapochynok**
Adding Homework 4

### CODE REVIEW

**Comment by juliapochynok** [comment]
Marged HW3 to master after the review was done, and then merged master to this branch for changes to be seen.

File: src/components/contact_page/ContactPage.jsx [hunk 1]
```diff
@@ -3,11 +3,57 @@ import './ContactPage.css';
 import Navigator from '../general/Navigator';
 import Footer from '../general/Footer';
 import Form from './Form';
+import { createStore } from "redux";
+import { Provider } from "react-redux";
+
+
+
+const initialState = {
```
Comment (pavlo-berezin) [inline src/components/contact_page/ContactPage.jsx:11]: why is this all inside of component?

File: src/components/contact_page/Form.jsx [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
+  //   name: '',
+  //   email: '',
+  //   subject: '',
+  //   message: '',
+  // }
 
   handleSubmit(e) {
     e.preventDefault()
-    const { name, email, subject, message } = this.state
+    const name = this.props.name;
```
Comment (pavlo-berezin) [inline src/components/contact_page/Form.jsx:19]: `const { name, email, message, subject } = this.props;`

File: src/components/contact_page/ContactPage.jsx [hunk 3]
```diff
@@ -3,11 +3,57 @@ import './ContactPage.css';
...
+    name: '',
+    email: '',
+    subject: '',
+    message: '',
+}
+
+function reducer(state = initialState, action) {
+switch(action.type) {
```
Comment (pavlo-berezin) [inline src/components/contact_page/ContactPage.jsx:19]: It's not really good to use Redux for just a form. But lets say it's okay in your case, cause you app doesn't really need it.

File: src/components/contact_page/Form.jsx [hunk 4]
```diff
@@ -34,18 +43,33 @@ export default class Form1 extends Component {
...
+    //   email: '',
+    //   subject: '',
+    //   message: '',
+    // })
+    this.props.dispatch({ type: "RESET"});
   }
 
-  handleChange = (param, e) => {
-      this.setState({ [param]: e.target.value })
+  handleChangeName = (param, e) => {
```
Comment (pavlo-berezin) [inline src/components/contact_page/Form.jsx:55]: you could just do:
```
handleChange = (param, e) => {
  this.props.dispatch({ type: param, arg: e.target.value });
}
```

**Pull request merged**